### PR TITLE
Use unfiltered DAO for summary view

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-2562

## Changes
- Add unfiltered DAO to dao config and use it on DAO summary view
- Always fetch data on double clicking a table view row

### CODE REVIEW

File: src/foam/u2/view/UnstyledTableView.js [hunk 1]
```diff
@@ -455,7 +455,7 @@ foam.CLASS({
                   }).
                   callIf(view.dblclick && ! view.disableUserSelection, function() {
                     tableRowElement.on('dblclick', function() {
-                      view.dblclick && view.dblclick(canObjBeBuildFromProjection ? obj : null, obj.id);
+                      view.dblclick && view.dblclick(null, obj.id);
```
Comment (olhabn) [inline src/foam/u2/view/UnstyledTableView.js:458]: but if canObjBeBuildFromProjection is false, then we have a "full" object here. why do we need to look it up again?

Comment (lchanmann) [reply src/foam/u2/view/UnstyledTableView.js:458]: we discussed yesterday to always load the object regardless, wasn't it? @kgrgreer

Comment (olhabn) [reply src/foam/u2/view/UnstyledTableView.js:458]: oh I see. I missed this part

File: src/foam/comics/v2/DAOControllerConfig.js [hunk 2]
```diff
@@ -51,6 +51,16 @@ foam.CLASS({
         return dao;
       }
     },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'unfilteredDAO',
+      hidden: true,
+      expression: function(dao) {
+        if ( dao && foam.dao.FilteredDAO.isInstance(dao) ) {
```
Comment (jlhughes) [inline src/foam/comics/v2/DAOControllerConfig.js]: This seems very fragile.

Comment (kgrgreer) [inline src/foam/comics/v2/DAOControllerConfig.js]: Shouldn't we return just 'dao' in the else case?

File: src/foam/comics/v2/DAOSummaryView.js [hunk 3]
```diff
@@ -222,7 +222,8 @@ foam.CLASS({
       var self = this;
       this.SUPER();
 
-      var promise = this.data ? Promise.resolve(this.data) : this.config.dao.inX(this.__subContext__).find(this.id);
+      var dao = this.config.unfilteredDAO || this.config.dao;
```
Comment (kgrgreer) [inline src/foam/comics/v2/DAOSummaryView.js]: Then we wouldn't need this ||.

Comment (lchanmann) [reply src/foam/comics/v2/DAOSummaryView.js]: Fixed in https://github.com/foam-framework/foam2/pull/4195/commits/88f83f6a3046f5c5dcd0653fb2ddb432398b08b9.

File: src/foam/comics/v2/DAOControllerConfig.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/comics/v2/DAOControllerConfig.js]: Fixed in https://github.com/foam-framework/foam2/pull/4195/commits/88f83f6a3046f5c5dcd0653fb2ddb432398b08b9.

Comment (lchanmann) [reply src/foam/comics/v2/DAOControllerConfig.js]: Fixed in https://github.com/foam-framework/foam2/pull/4195/commits/899ba227eebfeb7fbfc0eefdfafd27974b50c25e.

File: src/foam/comics/v2/DAOControllerConfig.js [hunk 4]
```diff
@@ -51,6 +51,21 @@ foam.CLASS({
...
     },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'unfilteredDAO',
+      hidden: true,
+      expression: function(dao) {
+        var result = dao;
+        while ( dao && foam.dao.ProxyDAO.isInstance(dao) ) {
+          if ( foam.dao.FilteredDAO.isInstance(dao) ) {
+            result = dao.delegate;
```
Comment (kgrgreer) [inline src/foam/comics/v2/DAOControllerConfig.js]: Simpler to just return dao.delegate?

Comment (lchanmann) [reply src/foam/comics/v2/DAOControllerConfig.js]: To not assume there is only one filtered dao in the chain, the real unfiltered dao is the delegate of the last filtered dao.

Comment (lchanmann) [reply src/foam/comics/v2/DAOControllerConfig.js]: Fixed in https://github.com/foam-framework/foam2/pull/4195/commits/a8d74914044caa6d22c810e3a63ff1bded612689.

**Pull request merged**